api-parent/access-api/src/main/java/com/sap/cloud/environment/servicebinding/api/ServiceBindingAccessor.java

```diff
@@ -16,6 +19,37 @@
 @FunctionalInterface
 public interface ServiceBindingAccessor
 {
+    /**
+     * Returns {@link ServiceBindingAccessor} instances for implementations that are exposed via the
+     * <a href="https://xperti.io/blogs/understanding-java-service-loader/">Service Loader Pattern</a>.
+     * <p>
+     * </p>
+     * These instances are useful when the behavior of one (or more) specific {@link ServiceBindingAccessor}s should be
+     * overwritten while leaving others in their default state. <br>
+     * <b>Example:</b>
+     * 
+     * <pre>
+     * final List<ServiceBindingAccessor> defaultInstances = ServiceBindingAccessor.getInstancesViaServiceLoader();
+     * if( defaultInstances.removeIf(SapVcapServicesServiceBindingAccessor.class::isInstance) ) {
+     *     defaultInstances.add(new SapVcapServicesServiceBindingAccessor(customEnvironmentVariableReader));
+     * }
+     *
+     * final ServiceBindingMerger merger = new ServiceBindingMerger(defaultInstances, ServiceBindingMerger.KEEP_UNIQUE);
+     * final SimpleServiceBindingCache cache = new SimpleServiceBindingCache(merger);
+     *
+     * DefaultServiceBindingAccessor.setInstance(cache);
+     * </pre>
+     *
+     * @return A {@link List} of {@link ServiceBindingAccessor} instances created from implementations that are exposed
+     *         via the <i>Service Locator Pattern</i> (see above).
+     */
+    static List<ServiceBindingAccessor> getInstancesViaServiceLoader()
```

Comment:
Would it make more sense to go with a `Collection` here?
From my understanding the ServiceLoader doesn't provide any guarantees on the order of the elements returned, so returning a list might give the wrong impression.

Reply:
That's a very valid remark.

However, I consciously chose `List` over `Collection` to keep the APIs as consistent as possible.
For example, the `ServiceBindingAccessor#getServiceBindings` also returns a `List` while is actually does not guarantee any specific ordering (nevertheless there **is** an order, so you can access items by index).
IMO the same semantic applies here.

Reply:
Hm, I'm not sure how I feel about this topic 😅
In general `List` is kind of the standard collection interface, which is the easiest to work with.
However, indexed access doesn't really happen too often (how would you know what accessor would be in `get(2)`? The more common use case would be to iterate over the returned collection, which Collection (or any `Iterable`) supports out of the box (allowing for the `for(ServiceBindingAccessor accessor : getInstancesViaServiceLoader)` loop).

In the end this doesn't really make any big difference.

Reply:
> In the end this doesn't really make any big difference.

Yeah, I thought so too. Therefore, I decided to use `List` pretty much everywhere just to be consistent.